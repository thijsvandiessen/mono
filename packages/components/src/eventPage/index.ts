export { EventPage } from './eventPage.jsx'
