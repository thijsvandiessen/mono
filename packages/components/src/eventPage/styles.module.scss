$image-width: 100%;
$image-height: calc($image-width * 1.414);
$padding-left-sidebar: 3rem;
$shadow: 5px 5px 20px 5px rgba(0, 0, 0, 0.1);

.container {
  display: grid;
  grid-template-areas:
    'main'
    'sidebar';
  grid-template-columns: 1fr;

  @media screen and (min-width: 50rem) {
    grid-template-columns: 33% 1fr;
    grid-template-areas: 'sidebar main';
  }
}

.noGrid {
  grid-template-areas:
    'main'
    'sidebar';
  grid-template-columns: 1fr;
}

.header {
  grid-area: header;

  background-color: var(--colour-primary);
  color: var(--colour-secondary-light);
  padding: 4rem 0;

  @media screen and (min-width: 50rem) {
    padding: 6rem 0;
  }
}

.program {
  list-style: none;
  padding: 0;
  margin: 0;
  margin-top: 1rem;

  @media screen and (min-width: 50rem) {
    text-align: center;
  }
}

.programItem {
  color: var(--colour-background);
}

.programTitle {
  font-weight: bold;
}

.sidebar {
  grid-area: sidebar;
  margin-bottom: var(--layout-default-mobile-spacing);

  @media screen and (min-width: 50rem) {
    margin-bottom: 0;
  }
}

$sidebar-translate: 50px;

.sidebarContent {
  @media screen and (min-width: 50rem) {
    position: sticky;
    top: calc(0.5rem + $sidebar-translate);
    transform: translateY(calc($sidebar-translate * -1));
  }
}

.content {
  grid-area: main;

  @media screen and (min-width: 50rem) {
    padding-left: $padding-left-sidebar;
  }
}

.imageContainer {
  border-radius: 8px;
  flex-shrink: 0;
  max-width: fit-content;

  @media screen and (min-width: 50rem) {
    box-shadow: $shadow;
  }
}

.image {
  width: $image-width;
  height: $image-height;
  max-width: 100%;
  border-radius: inherit;
  object-fit: cover;
}

.persons {
  list-style: none;
  padding: 0;
  margin: 0;
  background-color: var(--colour-background);
  border-radius: 8px;

  @media screen and (min-width: 50rem) {
    box-shadow: $shadow;
    padding: 2rem 2rem;
  }
}

.person:not(:last-child) {
  margin-bottom: 1rem;

  @media screen and (min-width: 50rem) {
    margin-bottom: 2rem;
  }
}

.personTitle {
  text-transform: uppercase;
}

.imageContainer + .persons {
  margin-top: 1rem;

  @media screen and (min-width: 50rem) {
    margin-top: 3rem;
  }
}

.eventListItem {
  margin-top: var(--layout-default-mobile-spacing);

  @media screen and (min-width: 50rem) {
    margin-left: 1.5rem;
    margin-top: var(--layout-default-desktop-spacing);
  }
}

.root .pageContent {
  margin-top: calc(var(--layout-default-mobile-spacing) - 1.5rem);

  @media screen and (min-width: 50rem) {
    margin-top: calc(var(--layout-default-desktop-spacing) - 2rem);
  }
}

.locations {
  grid-area: footer;
  display: flex;
  flex-direction: column;
  gap: 2rem;

  @media screen and (min-width: 50rem) {
    flex-direction: row;
    gap: 0;
  }
}
