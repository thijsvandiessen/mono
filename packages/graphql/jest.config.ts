import { type Config } from 'jest'

const config: Config = {
  // Add more setup options before each test is run
  setupFilesAfterEnv: ['./jest.setup.cjs'],
  testEnvironment: 'jest-environment-jsdom',
  preset: 'ts-jest',
  transform: {
    '^.+\\.tsx?$': [
      'ts-jest',
      {
        babel: true,
        tsconfig: 'tsconfig.test.json',
      },
    ],
  },
  collectCoverage: true,
  coverageDirectory: 'coverage',
}

export default config
