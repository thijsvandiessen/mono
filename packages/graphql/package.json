{
  "name": "@mono/graphql",
  "version": "1.0.0",
  "description": "",
  "type": "module",
  "exports": {
    ".": {
      "import": "./src/index.ts",
      "module": "./dist/graphql.js",
      "types": "./dist/graphql.d.ts"
    }
  },
  "main": "./dist/graphql.js",
  "types": "./dist/graphql.d.ts",
  "files": [
    "./dist/*"
  ],
  "scripts": {
    "build": "vite build",
    "codegen": "graphql-codegen --config ./codegen.ts",
    "codegen:watch": "graphql-codegen --require dotenv/config --config ./codegen.ts --watch",
    "download-schema": "rover graph fetch vandiessen-mono-graph@current > ./src/generated/schema.graphql && npm run lint:styles",
    "lint": "npm run lint:js && npm run lint:styles && npm run lint:typescript",
    "lint:js": "eslint . --fix",
    "lint:styles": "prettier . --write --cache --log-level silent --list-different",
    "lint:typescript": "tsc",
    "test": "jest"
  },
  "dependencies": {
    "@mono/utils": "1.0.0",
    "@urql/core": "5.0.6",
    "@urql/exchange-graphcache": "7.1.2",
    "@urql/exchange-retry": "1.3.0",
    "graphql": "16.9.0",
    "react": "18.3.1"
  },
  "devDependencies": {
    "@apollo/rover": "0.26.2",
    "@commitlint/config-conventional": "19.2.2",
    "@graphql-codegen/cli": "5.0.2",
    "@graphql-codegen/client-preset": "4.3.3",
    "@graphql-codegen/introspection": "4.0.3",
    "@graphql-codegen/typescript": "4.0.9",
    "@graphql-codegen/typescript-document-nodes": "4.0.9",
    "@graphql-codegen/typescript-operations": "4.2.3",
    "@testing-library/jest-dom": "6.5.0",
    "@testing-library/react": "16.0.1",
    "@types/jest": "29.5.12",
    "@types/node": "22.5.4",
    "@types/react": "18.3.5",
    "@types/react-dom": "18.2.18",
    "@typescript-eslint/eslint-plugin": "8.5.0",
    "@typescript-eslint/parser": "8.5.0",
    "dotenv": "16.4.5",
    "eslint": "8.57.0",
    "eslint-config-next": "14.2.8",
    "eslint-plugin-react-hooks": "4.6.2",
    "eslint-plugin-testing-library": "6.3.0",
    "jest": "29.7.0",
    "jest-environment-jsdom": "29.7.0",
    "prettier": "3.3.3",
    "pretty-quick": "4.0.0",
    "turbo": "2.1.1",
    "typescript": "5.5.4"
  }
}
