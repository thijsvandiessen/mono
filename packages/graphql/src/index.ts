export { defaultExchange } from './cacheExchange'
export { client } from './gqlClient'
export { linkConfig } from './linkConfig'
export {
  getEvent,
  getEventPage,
  getEventPageSeo,
  getEvents,
  getEventsMeta,
  getEventUrls,
  getFutureEvents,
  getGeneralInfo,
  getLocation,
  getLocations,
  getPage,
  getPages,
  getPageSeo,
  getPagesMeta,
  getSiteInfo,
  getSiteMetadata,
} from './getters'

export type {
  Author,
  BlogPost,
  Event,
  PageLink,
  CloudinaryAsset,
  Image,
  Location,
  Music,
} from './types'

export { isOfTypeCloudinaryAsset } from './types'
export { formatCloudinaryDocument, formatCloudinaryImage } from './formatters'
export * from './generated/graphql'
