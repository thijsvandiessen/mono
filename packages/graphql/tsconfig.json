{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "lib": ["ESNext", "DOM"]
  },
  "include": ["src/**/*", "src/*"],
  "exclude": ["src/**/*.test.ts", "src/**/*.test.tsx"]
}
