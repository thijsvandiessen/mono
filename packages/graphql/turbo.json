{
  "extends": ["//"],
  "pipeline": {
    "lint": {},
    "test": {},
    "lint:js": {},
    "lint:styles": {},
    "lint:typescript": {}
  },
  "remoteCache": {
    "signature": false
  },
  "globalDependencies": ["tsconfig.json"]
}
