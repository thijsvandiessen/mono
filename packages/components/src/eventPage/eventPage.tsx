import React from 'react'
import Image from 'next/image'
import { notFound } from 'next/navigation'
import classNames from 'classnames'
import { PageContent } from '../pageContent'
import { LocationDetail } from '../locationDetail'
import { EventListItem } from '../eventListItem'
import { getEventPage } from '@mono/graphql'

import styles from './styles.module.scss'

export interface Props {
  slug: string
}

export const EventPage = async ({ slug }: Props) => {
  const { data } = await getEventPage({ slug })
  if (!data) return notFound()

  return (
    <article className={styles.root}>
      <header className={classNames(styles.header)}>
        <div className={classNames(styles.titleContainer, 'content-layout')}>
          <h2>{data.title}</h2>
        </div>
      </header>
      <div
        className={classNames(styles.container, 'content-layout', {
          [`${styles.noGrid}`]: !data.image?.url,
        })}
      >
        <aside className={styles.sidebar}>
          <div className={styles.sidebarContent}>
            {data.image?.url && (
              <div className={classNames(styles.imageContainer)}>
                <Image
                  priority={true} // TODO: only for the first item
                  className={classNames(styles.image)}
                  src={data.image.url}
                  alt={data.image.description}
                  width={data.image.width ?? 100}
                  height={data.image.height ?? 100}
                />
              </div>
            )}
          </div>
        </aside>
        <div className={styles.content}>
          {data.locations.length > 0 && (
            <div className={styles.eventListItem}>
              <EventListItem
                data={data}
                size="large"
                isLast={true}
                title="Agenda"
                showLink={false}
                showImage={false}
                showProgram={false}
              />
            </div>
          )}

          <PageContent
            sectionClassName={styles.pageContent}
            items={data.content}
          />
        </div>
      </div>
      <div className={styles.locations}>
        {data.locations.map((location) => {
          if (!location?.id || !location.startTime) return null
          return (
            <LocationDetail
              key={location?.id}
              id={location.id}
              startTime={location.startTime}
            />
          )
        })}
      </div>
    </article>
  )
}
