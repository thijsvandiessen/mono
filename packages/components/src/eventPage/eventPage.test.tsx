import React from 'react'
import { EventPage } from './eventPage'
import type { Props } from './eventPage'
import { getEventPage } from '@mono/graphql'
import { resolvedComponent } from '@mono/utils'
import { render } from '@testing-library/react'

jest.mock('../navigation', () => {
  const originalModule = jest.requireActual('../navigation')
  return {
    __esModule: true,
    ...originalModule,
    Navigation: jest.fn(() => <div>[Navigation]</div>),
  }
})

jest.mock('@mono/graphql', () => {
  const originalModule = jest.requireActual(
    '@mono/graphql'
  )
  return {
    __esModule: true,
    ...originalModule,
    getEventPage: jest.fn(),
  }
})

const getEventPageMock = jest.mocked(getEventPage)

describe('Concert component', () => {
  it('shows all the data', async () => {
    getEventPageMock.mockResolvedValue({
      data: {
        id: 'unique-id',
        title: 'title',
        image: undefined,
        locations: [],
        url: '/some-slug',
        content: [],
      },
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(EventPage, {
      slug: 'tournee-20',
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()
  })
})
