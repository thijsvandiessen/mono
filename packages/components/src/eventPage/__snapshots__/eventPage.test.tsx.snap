// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Concert component > shows all the data 1`] = `
<div>
  <article
    class="_root_3155da"
  >
    <header
      class="_header_3155da"
    >
      <div
        class="_titleContainer_3155da content-layout"
      >
        <h1>
          title
        </h1>
      </div>
    </header>
    <div
      class="_container_3155da content-layout _noGrid_3155da"
    >
      <aside
        class="_sidebar_3155da"
      >
        <div
          class="_sidebarContent_3155da"
        />
      </aside>
      <div
        class="_content_3155da"
      />
    </div>
    <div
      class="_locations_3155da"
    />
  </article>
</div>
`;
